store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java

```diff
@@ -481,9 +482,13 @@ private void buildTableSchema(Field[] fields, TableSchemaBuilder tableSchemaBuil
           ColumnSchema columnSchema = tableSchemaBuilder
               .addColumn(new StructField(field.getFieldName(), field.getDataType()),
                   valIndex, isSortColumn > -1);
-          columnSchema.setSortColumn(true);
           if (isSortColumn > -1) {
+            columnSchema.setSortColumn(true);
             sortColumnsSchemaList[isSortColumn] = columnSchema;
+          } else if (sortColumnsList.isEmpty() && columnSchema.isDimensionColumn()) {
```

Comment:
Complex column and it's children should not be considered as sort column, please handle the same

Reply:
complex check is already handled above.